src/kubernetes.cc

```diff
@@ -1042,27 +1042,43 @@ json::value KubernetesReader::FindTopLevelOwner(
 #endif
   const json::Object* metadata = obj->Get<json::Object>("metadata");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: metadata is " << *metadata;
+  LOG(DEBUG) << "FindTopLevelController: metadata is " << *metadata;
 #endif
   if (!metadata->Has("ownerReferences")) {
 #ifdef VERBOSE
-    LOG(DEBUG) << "FindTopLevelOwner: no owner references in " << *metadata;
+    LOG(DEBUG) << "FindTopLevelController: no owner references in " << *metadata;
 #endif
     return object;
   }
   const json::Array* refs = metadata->Get<json::Array>("ownerReferences");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: refs is " << *refs;
+  LOG(DEBUG) << "FindTopLevelController: refs is " << *refs;
 #endif
-  if (refs->size() > 1) {
+  std::vector<json::Object> controller_refs;
```

Comment:
Let's not copy JSON values. Given that the `refs` array will live to the end of this function, I would suggest just storing `const json::Object` pointers in the array.

Reply:
done